web/src/components/Toast.tsx

```diff
@@ -1,82 +1,131 @@
 import * as ToastPrimitive from '@radix-ui/react-toast';
-import { X } from 'lucide-react';
+import { CircleCheck, Info, OctagonX, TriangleAlert, X } from 'lucide-react';
 import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { twMerge } from 'tailwind-merge';
 
 import { Button } from './Button';
 
-type Props = {
-  title: string;
-  description?: string;
+export enum ToastType {
+  INFO = 'info',
+  SUCCESS = 'success',
+  WARNING = 'warning',
+  DANGER = 'danger',
+}
+
+interface ToastProps {
+  title?: string;
+  description: string;
   isCloseable?: boolean;
   toastAction?: () => void;
   toastClose?: () => void;
   toastActionText?: string;
   toastCloseText?: string;
   duration?: number;
+  type?: ToastType;
+}
+
+export const ToastTypeTheme = {
+  [ToastType.SUCCESS]: {
+    Icon: CircleCheck,
+    color:
+      'text-emerald-800 dark:text-emerald-500 before:bg-emerald-800 dark:before:bg-emerald-500',
+  },
+  [ToastType.WARNING]: {
+    Icon: TriangleAlert,
+    color:
+      'text-amber-700 dark:text-amber-500 before:bg-amber-700 dark:before:bg-amber-500',
+  },
+  [ToastType.DANGER]: {
+    Icon: OctagonX,
+    color: 'text-red-700 dark:text-red-400 before:bg-red-700 dark:before:bg-red-400',
+  },
+  [ToastType.INFO]: {
+    Icon: Info,
+    color: 'text-blue-800 dark:text-blue-400 before:bg-blue-800 dark:before:bg-blue-400',
+  },
 };
 
-function Toast({
+export function Toast({
   title,
   description,
   toastAction,
   toastActionText,
   toastClose,
   toastCloseText,
   duration,
-}: Props) {
+  type = ToastType.INFO,
+}: ToastProps) {
   const { t } = useTranslation();
   const [open, setOpen] = useState(true);
   const handleToastAction = () => {
     toastAction?.();
     setOpen(false);
   };
+
   const handleToastClose = () => {
     toastClose?.();
     setOpen(false);
   };
 
+  const { Icon, color } = ToastTypeTheme[type];
+
   return (
     <>
       <ToastPrimitive.Root
+        data-testid="toast"
         open={open}
         onOpenChange={setOpen}
         duration={duration}
         type="background"
-        className="fixed top-16 z-50 m-2 inline-block max-w-max self-center rounded-lg bg-white/80 shadow backdrop-blur-sm dark:bg-gray-800/80 xs:left-1/2 xs:w-max xs:-translate-x-1/2 xs:transform"
+        className={twMerge(
+          'fixed left-1/2 top-16 z-50 inline-block flex w-11/12 min-w-96 max-w-full -translate-x-1/2 transform rounded-lg shadow xs:w-max',
```

Comment:
The min-w causes some issues on iPhone SE: 
<img width="487" alt="image" src="https://github.com/user-attachments/assets/432cada8-057a-4159-98e5-1646b0ff2fa1">
and flex and inline-block apply the same properties. Also I haven't seen us use xs before, is this required?
```suggestion
          'fixed left-1/2 top-16 z-50 flex w-11/12  max-w-full -translate-x-1/2 transform rounded-lg shadow xs:w-max',
```

Reply:
It's always the SE...

Reply:
@tonypls fixed the dynamic width without xs, if you want to take another look